javalib/src/main/scala/java/util/Date.scala

```diff
@@ -33,6 +37,32 @@ class Date(var milliseconds: Long)
   def setTime(time: Long): Unit =
     milliseconds = time
 
-  override def toString(): String = s"Date($milliseconds)"
+  override def toString(): String = {
+    val seconds = milliseconds / 1000L
+    val default = s"Date($milliseconds)"
+    Date.secondsToString(seconds, default)
+  }
+}
+
+private object Date {
+  def secondsToString(seconds: Long, default: String): String = Zone {
+    implicit z =>
+      val ttPtr = alloc[time_t]
+      !ttPtr = seconds
 
+      val tmPtr = alloc[tm]
+
+      if (localtime_r(ttPtr, tmPtr) == null) {
+        default
+      } else {
+        // 40 is over-provisioning.
+        // Most result strings should be about 28 + 1 for terminal NULL
+        // + 2 because some IANA timezone abbreviation can have 5 characters.
+        val bufSize = 40
+        val buf     = alloc[Byte](bufSize)
+        val n       = strftime(buf, bufSize, c"%a %b %d %T %Z %Y", tmPtr)
```

Comment:
I've tried to test this code locally in sandbox after seeing problems in CI. As long as I had no env variable `TZ` set it ended up in SeqFault. After setting it up this code worked correctly. Also it worked after switching to `%z" (numeric timezone).

Reply:
@WojciechMazur Thank you for the effort you put into this.

I think that you are seeing the "will fail" portion of the WIP message above:
`This WIP PR requires PR #2066 and DateTest will fail until that PR has been merged.`

PR #2066, which I know that you have read, changes the way Scala Native instances of `tm`
type gets translated to C `struct tm`.  It touches only POSIX defined fields, and then only
in an order-independent manner. You caught a bug in the PR ;-)   

What happens when the "wrong" sized structure is passed to C depends upon
many factors. Among them are the Scala Native optimization level. `release-fast`
tends to be slayed easily.  There can also be differences due to the Operating
System.  MacOS (spelling, name?) tends to break.

What do you see if you apply PR #2066 privately?

I believe that ``%z"` works with `wrong` size structures because
it does not touch the `missing` field (I think it is tzname). Missing
because 64 bit linux has it and it is used by glibc, as you discovered.)

Setting TZ works in a similar manner: Ask me no questions, I will tell
you no lies (SEGFAULT).

Reply:
@WojciechMazur   Based on the helpful discussion, I went back to an issue
that I had waffled on in the first generation.   I added a call to `tzset()` just
before `localtime_r`.  `localtime` is defined to act as if it had called `tzset()`,
localtime_r is not.

My reluctance had been the performance hit of the `tzset` call on each and
every `Date.toString()` call.    However, short of changing the SN bootstrap
code, there is no way I know to ensure that `tzset` has been called before
`localtime_r`.  A lazy variable will not do, because someone can change
the TZ variable.  I believe that `localtime` will pick up such a change.

For now, I go with correctness. If the performance of `Date.toString`
becomes an issue, the `tzset` call is a good place to look first.

PS:  I believe that we do not want to change the SN bootstrap
        code to call `tzset()` because more programs probably 
        never use it than those which do.